mne/io/snirf/_snirf.py

```diff
@@ -14,10 +14,58 @@
 from ..._freesurfer import get_mni_fiducials
 from ...annotations import Annotations
 from ...transforms import _frame_to_str, apply_trans
-from ...utils import _check_fname, _import_h5py, fill_doc, logger, verbose, warn
+from ...utils import (
+    NamedInt,
+    _check_fname,
+    _check_option,
+    _import_h5py,
+    fill_doc,
+    logger,
+    verbose,
+    warn,
+)
 from ..base import BaseRaw
 from ..nirx.nirx import _convert_fnirs_to_head
 
+SNIRF_CW_AMPLITUDE = NamedInt("SNIRF_CW_AMPLITUDE", 1)
+SNIRF_TD_GATED_AMPLITUDE = NamedInt("SNIRF_TD_GATED_AMPLITUDE", 201)
+SNIRF_TD_MOMENTS_AMPLITUDE = NamedInt("SNIRF_TD_MOMENTS_AMPLITUDE", 301)
+SNIRF_PROCESSED = NamedInt("SNIRF_PROCESSED", 99999)
+_AVAILABLE_SNIRF_DATA_TYPES = (
+    SNIRF_CW_AMPLITUDE,
+    SNIRF_TD_GATED_AMPLITUDE,
+    SNIRF_TD_MOMENTS_AMPLITUDE,
+    SNIRF_PROCESSED,
+)
+
+
+# SNIRF: Supported measurementList(k).dataTypeLabel values in dataTimeSeries
+FNIRS_SNIRF_DATATYPELABELS = {
+    # These types are specified here:
+    # https://github.com/fNIRS/snirf/blob/master/snirf_specification.md#supported-measurementlistkdatatypelabel-values-in-datatimeseries  # noqa: E501
+    "HbO": 1,  # Oxygenated hemoglobin (oxyhemoglobin) concentration
+    "HbR": 2,  # Deoxygenated hemoglobin (deoxyhemoglobin) concentration
+    "HbT": 3,  # Total hemoglobin concentration
+    "dOD": 4,  # Change in optical density
+    "mua": 5,  # Absorption coefficient
+    "musp": 6,  # Scattering coefficient
+    "H2O": 7,  # Water content
+    "Lipid": 8,  # Lipid concentration
+    "BFi": 9,  # Blood flow index
+    "HRF dOD": 10,  # HRF for change in optical density
+    "HRF HbO": 11,  # HRF for oxyhemoglobin concentration
+    "HRF HbR": 12,  # HRF for deoxyhemoglobin concentration
+    "HRF HbT": 13,  # HRF for total hemoglobin concentration
+    "HRF BFi": 14,  # HRF for blood flow index
+}
+
+# In each file, the TD moment order maps to these values
+_TD_MOMENT_ORDER_MAP = {
+    0: "intensity",
+    1: "mean",
+    2: "variance",
+}
```

Comment:
@julien-dubois-k I just took a stab at this, not sure if it's actually correct?

Reply:
yes